config/routes.rb

```diff
@@ -809,6 +809,7 @@ def patch(*) end # Disable PATCH requests
 
     # Topic routes
     get "t/id_for/:slug" => "topics#id_for_slug"
+    get "t/external_id/:external_id" => "topics#show_by_external_id", format: :json, constrains: { external_id: /\A[\w-]+\z/ }
```

Comment:
@oblakeerickson one of the options is invalid: `constrains` => `constraints` 🙂 (better late than never 😛)